bokeh/tests/test_palettes.py

```diff
@@ -1,4 +1,440 @@
-from bokeh.palettes import viridis, Viridis10
+import re
+from types import FunctionType
 
-def test_cmap_generator_function():
-    assert viridis(10) == Viridis10
+import bokeh.palettes as pal
+
+# Keys for Natural Sorting
+def ret_ns_key (string):
+    num = re.findall('\d+',string)[0]
+    return [string[:-len(num)], int(num)]
+
+def test_for_palettes():
+    palettes = [x for x in dir(pal) if isinstance(getattr(pal,x),list)]
+    # __palettes__ is also return and should not be tested
+    palettes = palettes[:-1]
+    assert sorted(palettes, key=ret_ns_key) == sorted(pal.__palettes__, key=ret_ns_key)
+
+def test_palette_lengths():
+    palettes = [x for x in dir(pal) if isinstance(getattr(pal,x),list)]
+    # __palettes__ is also return and should not be tested
+    palettes = palettes[:-1]
+    for palstr in palettes:
+        assert len(getattr(pal,palstr)) == int(re.findall('\d+',palstr)[0])
+
+def test_cmap_generator_functions():
```

Comment:
While this "works", it's hard to read and will be a pain to update in the future. Generally, it's a good strategy to stay away from string manipulation if possible. I'd recommend something like:

```python
funcs = [x for x in dir(pal) if isinstance(getattr(pal, x), FunctionType)]
# First value is _cmap_func_generator and that should not be directly tested
funcs = funcs[1:]
cmap_functions = {viridis: Viridis256, gray: Grey256, ...}
# check that you're testing all of the cmap funcs (and that there aren't any extra)
assert len(set(funcs).symmetric_difference(cmap_functions.keys())) == 0
# check that they all create what's expected
for k, v in cmap_functions.items():
    assert k(256)==v
```